File: 1983-Widest Pair of Indices With Equal Range Sum.py
class Solution(object):
    def widestPairOfIndices(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: int
        """
        prefixAll = [0]
        for i in range(len(nums1)):
            prefixAll.append(nums1[i] - nums2[i] + prefixAll[-1])
        
        res = 0
        hmap = dict()
        for i in range(len(prefixAll)):
            if prefixAll[i] not in hmap:
                hmap[prefixAll[i]] = i
            else:
                res = max(res, i-hmap[prefixAll[i]])
        
        return res

File: JavaScript/1983-Widest Pair of Indices With Equal Range Sum.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var widestPairOfIndices = function(nums1, nums2) {
    let prefixAll = [0];
    const N = nums1.length
    for (let i=0; i<N; i++){
        prefixAll.push(nums1[i]-nums2[i]+prefixAll[prefixAll.length-1]);
    }
    
    let res = 0, hmap = new Map();
    for (let i=0; i<N+1; i++){
        if ( !hmap.has(prefixAll[i]) ){
            hmap.set(prefixAll[i], i);
        }else{
            res = Math.max(res, i-hmap.get(prefixAll[i]) );
        }
    }
    return res;
};